pkg/virtctl/create/instancetype/instancetype.go

```diff
@@ -0,0 +1,299 @@
+	var policy v1.IOThreadsPolicy
+
+	switch c.ioThreadsPolicy {
+	case string(v1.IOThreadsPolicyAuto):
+		policy = v1.IOThreadsPolicyAuto
+	case string(v1.IOThreadsPolicyShared):
+		policy = v1.IOThreadsPolicyShared
+	default:
+		return params.FlagErr(IOThreadsPolicyFlag, IOThreadErr)
+	}
+
+	instancetypeSpec.IOThreadsPolicy = &policy
+	return nil
+}
+
+func (c *createInstancetype) usage() string {
+	return `  # Create a manifest for a ClusterInstancetype with a random name:
+  {{ProgramName}} create instancetype --cpu 2 --memory 256Mi
+  
+  # Create a manifest for a ClusterInstancetype with a specified name:
+  {{ProgramName}} create instancetype --name my-instancetype --cpu 2 --memory 256Mi
+
+  # Create a manifest for a ClusterInstancetype with a specified gpu:
+  {{ProgramName}} create instancetype --cpu 2 --memory 256Mi --gpu name:gpu1,devicename:nvidia
+  
+  # Create a manifest for a Instancetype with a specified name and cpu:
+  {{ProgramName}} create instancetype --namespaced --name my-instancetype --cpu 2 --memory 256Mi
+  
+  # Create a manifest for a ClusterInstancetype and use it to create a resource with kubectl
+  {{ProgramName}} create instancetype --cpu 2 --memory 256Mi | kubectl create -f -`
+}
+
+func (c *createInstancetype) newInstancetype(cmd *cobra.Command) (*instancetypev1alpha2.VirtualMachineInstancetype, error) {
+	return &instancetypev1alpha2.VirtualMachineInstancetype{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "VirtualMachineInstancetype",
+			APIVersion: instancetypev1alpha2.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: c.name,
+		},
+		Spec: instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+			CPU: instancetypev1alpha2.CPUInstancetype{
+				Guest: c.cpu,
+			},
+			Memory: instancetypev1alpha2.MemoryInstancetype{
+				Guest: resource.MustParse(c.memory),
+			},
+		},
+	}, nil
+}
+
+func (c *createInstancetype) newClusterInstancetype(cmd *cobra.Command) (*instancetypev1alpha2.VirtualMachineClusterInstancetype, error) {
+	return &instancetypev1alpha2.VirtualMachineClusterInstancetype{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "VirtualMachineClusterInstancetype",
+			APIVersion: instancetypev1alpha2.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: c.name,
+		},
+		Spec: instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+			CPU: instancetypev1alpha2.CPUInstancetype{
+				Guest: c.cpu,
+			},
+			Memory: instancetypev1alpha2.MemoryInstancetype{
+				Guest: resource.MustParse(c.memory),
+			},
+		},
+	}, nil
+}
+
+func (c *createInstancetype) applyFlags(cmd *cobra.Command, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) error {
+	for flag := range optFns {
+		if cmd.Flags().Changed(flag) {
+			if err := optFns[flag](c, instancetypeSpec); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (c *createInstancetype) validateFlags() error {
+	if _, err := resource.ParseQuantity(c.memory); err != nil {
```

Comment:
Store the result in `createInstancetype` struct and use it when creating the instancetype?

Reply:
Storing memory in `createInstancetype` struct would complicate things, because `c.memory` is a string that we take from user and `Spec.Memory.Guest` is `resource.Quantity`.

Reply:
ACK